state/tx_filter.go

```diff
@@ -10,6 +11,7 @@ import (
 func TxPreCheck(state State) mempl.PreCheckFunc {
 	maxDataBytes := types.MaxDataBytesUnknownEvidence(
 		state.ConsensusParams.Block.MaxBytes,
+		crypto.BLS12381,
```

Comment:
Why is this change necessary?

Reply:
Its because signature sizes are different in BLS and ED keys, which will change the max data bytes.
BLS public keys are 48 bytes and sigs are 96 bytes. Ed public keys are 32 bytes and sigs are 64 bytes. So a 16 byte difference for a public key and a 32 byte difference for the signature.

`maxEvidenceBytes := int64(maxNumEvidence) * MaxEvidenceBytesForKeyType(keyType)
	maxDataBytes := maxBytes -
		MaxOverheadForBlock -
		MaxHeaderBytes -
		int64(valsCount)*MaxVoteBytesForKeyType(keyType) -
		maxEvidenceBytes`

The max evidence bytes should then vary by the difference in public key size and signature size, here 48.

The max vote size should vary by the signature size, here 32.

I wrote this a while ago so am not 100% sure of my logic. I guess I should have commented more.

Reply:
Makes sense! Yes, I'd add a small godoc on this explaining how/why.